File: src/top150/n242_valid_anagram/solution.go
package main

func isAnagram(s string, t string) bool {
	chars := make([]int32, 26)
	for _, c := range s {
		chars[c-'a']++
	}

	for _, c := range t {
		chars[c-'a']--
	}

	for _, v := range chars {
		if v != 0 {
			return false
		}
	}

	return true
}

func main() {
	println(isAnagram("anagram", "nagaram"))
}

File: src/top150/n242_valid_anagram/Solution.java
package top150.n242_valid_anagram;

public class Solution {

    public boolean isAnagram(String s, String t) {
        int[] arr1 = new int[26];
        for (int i = 0; i < s.length(); i++) {
            arr1[s.charAt(i) - 'a']++;
        }

        for (int i=0; i < t.length(); i++) {
            arr1[t.charAt(i) - 'a']--;
        }

        for (int i: arr1) {
            if (i != 0)
                return false;
        }


        return true;
    }

    public static void main(String[] args) {
        var s = new Solution();
        System.out.println(s.isAnagram("anagram", "nagaram"));
    }
}
